packages/itwin/reports-config-widget/src/widget/components/ReportHorizontalTile.tsx

```diff
@@ -0,0 +1,132 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+import { useEffect, useRef, useState } from "react";
+import React from "react";
+import type BulkExtractor from "./BulkExtractor";
+import { ExtractionStates, ExtractionStatus } from "./ExtractionStatus";
+import type { BeEvent } from "@itwin/core-bentley";
+import { STATUS_CHECK_INTERVAL } from "./Constants";
+import type { Report } from "@itwin/insights-client";
+import { ReportsConfigWidget } from "../../ReportsConfigWidget";
+import {
+  DropdownMenu,
+  IconButton,
+  MenuItem,
+} from "@itwin/itwinui-react";
+import {
+  SvgDelete,
+  SvgEdit,
+  SvgMore,
+} from "@itwin/itwinui-icons-react";
+import { HorizontalTile } from "./HorizontalTile";
+import type { HorizontalTileProps } from "./HorizontalTile";
+
+export interface ReportHorizontalTileProps extends Pick<HorizontalTileProps, "onClickTitle"> {
+  selected: boolean;
+  onSelectionChange: (reportId: string, controlPressed: boolean) => void;
+  bulkExtractor: BulkExtractor;
+  jobStartEvent: BeEvent<(reportId: string) => void>;
+  report: Report;
+  onClickDelete: () => void;
+  onClickModify: () => void;
+}
+
+export const ReportHorizontalTile = (props: ReportHorizontalTileProps) => {
+  const [jobStarted, setJobStarted] = useState<boolean>(false);
+  const [extractionState, setExtractionState] = useState<ExtractionStates>(ExtractionStates.None);
+  const interval = useRef<number>();
+
+  useEffect(() => {
+    props.jobStartEvent.addListener((startedReportId: string) => {
+      if (startedReportId === props.report.id) {
+        setExtractionState(ExtractionStates.Starting);
+        setJobStarted(true);
+      }
+    });
+
+    return function cleanup() {
```

Comment:
Turn it into an arrow function

Reply:
function changed to an arrow function. Should I even simplify it to: `return props.jobStartEvent.clear;` ?

Reply:
Yeah the simplification works.

Reply:
Doesn't `clear` remove all listeners? You need to remove only that one listener for this specific component instead of all the listeners.

Reply:
@MindaugasButkus is right. It does clear all listeners. I am not too familiar with `BeEvent`. I am assuming there is something like `removeListener()`?

Reply:
Yes, there should be something like that and you need to have the exact same function reference that you passed into the `addListener`.

Reply:
Listener is created before adding or removing it from `jobStartEvent`.